### PR TITLE
fix(medusa): improved /store/customers endpoints

### PR DESCRIPTION
**Pull request by srindom**
**What?**
- Adds support for updating customer metadata
- Adds `defaultRelations`, `defaultFields` to responses 
- Hides `password_hash` in response
- Adds integration tests for `POST /customers` and `POST /customers/:id`
- Adds new error type: `DUPLICATE_ERROR` to allow `error-handler.js` to send 409 responses.

**Why?**
- Improved API compatability
- Better security

**How?**
- Following same pattern as seen in other routes

**Testing**
- integration tests for creation and update of customer (including auth)
- updates unit tests to account for `defaultFields/Relations`

### CODE REVIEW

File: integration-tests/api/__tests__/store/customer.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+const { dropDatabase } = require("pg-god");
```
Comment (srindom) [inline integration-tests/api/__tests__/store/customer.js:1]: First time we have integration tests for `/store/customers` so mostly skeleton.

File: integration-tests/api/__tests__/store/customer.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  describe("POST /store/customers/:id", () => {
+    beforeEach(async () => {
+      const manager = dbConnection.manager;
+      await manager.insert(Customer, {
+        id: "test_customer",
+        first_name: "John",
+        last_name: "Deere",
+        email: "john@deere.com",
+        password_hash:
```
Comment (srindom) [inline integration-tests/api/__tests__/store/customer.js:90]: This hash corresponds to the password "test".

File: integration-tests/api/__tests__/store/customer.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    it("updates a customer", async () => {
+      const api = useApi();
+
+      const authResponse = await api.post("/store/auth", {
+        email: "john@deere.com",
+        password: "test",
+      });
+
+      const customerId = authResponse.data.customer.id;
+      const [authCookie] = authResponse.headers["set-cookie"][0].split(";");
```
Comment (srindom) [inline integration-tests/api/__tests__/store/customer.js:110]: We log in first and then keep the set-cookie around so we can use it in the subsequent request.

File: packages/medusa-core-utils/src/errors.js [hunk 4]
```diff
@@ -5,10 +5,11 @@
 export const MedusaErrorTypes = {
   /** Errors stemming from the database */
   DB_ERROR: "database_error",
+  DUPLICATE_ERROR: "duplicate_error",
```
Comment (srindom) [inline packages/medusa-core-utils/src/errors.js:8]: This is appropriate for errors that have to do with duplication in the database. E.g. when trying to insert a customer with an email that already corresponds to a record in the customer table.

**Review by srindom — COMMENTED** [review]
Annotations

File: packages/medusa-core-utils/src/errors.js (hunk 4 above)
Comment (olivermrbl) [reply packages/medusa-core-utils/src/errors.js:8]: This is really nice!

**Review by olivermrbl — APPROVED** [review]
Clean! Can you create an issue (potentially `good-first-issue`) for updating all places we throw an error on duplicates? 
Places that come to mind: creating variants, completing cart, product options

File: integration-tests/api/__tests__/store/customer.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+const path = require("path");
+const { Customer } = require("@medusajs/medusa");
+
+const setupServer = require("../../../helpers/setup-server");
+const { useApi } = require("../../../helpers/use-api");
+const { initDb } = require("../../../helpers/use-db");
+
+const customerSeeder = require("../../helpers/customer-seeder");
+
+jest.setTimeout(30000);
```
Comment (VilfredSikker) [inline integration-tests/api/__tests__/store/customer.js:11]: Why is this necessary?

**Review by VilfredSikker — COMMENTED** [review]
Looks nice! I just have a single question:

File: integration-tests/api/__tests__/store/customer.js (hunk 5 above)
Comment (VilfredSikker) [reply integration-tests/api/__tests__/store/customer.js:11]: @srindom
